### DISCUSSION TITLE
FIBO usage

### DISCUSSION THREAD

**Discussion by rthomaskelly** [edmcouncil/fibo#1920] [General]
Hi I was looking into FIBO and thought it was really cool. I went ahead and loaded the ontology into neo4j to look and play. I wanted to make sure I'm doing it right? I was looking at your EquitiesExampleIndividuals file which has some dummy data. I tried to look at Apple only.

What graph DB do you guys use? All the FIBO refs tallied to like 20,000 nodes which was way too much for neo4j to render holistically. neo4j maybe doesn't render the namespaces nicely forcing ns__ onto properties? What else do you do with your data?

Apple example data nodes as Rdfs Labels and Skos Definitions, not sure which is right?

By Rdfs Label
![GraphAppleRdfsLabel](https://github.com/edmcouncil/fibo/assets/24437260/c940a643-ecc8-47b4-8196-28fa9e30057b)

By Skos Definition
![GraphAppleSkosDefinition](https://github.com/edmcouncil/fibo/assets/24437260/f62e9ff6-c26a-48af-abf1-3d7f387fd8a8)

**Comment by ElisaKendall** [comment]
Hi Thomas,

You should be able to load many more than 20K nodes in neo4j, and query them, though the desktop browser may not support that.  I would use the ontology element name rather than an annotation to name the nodes, or alternatively the rdfs:label rather than the skos:definition if you decide not to use that actual element name (i.e., the local name part of the IRI).

Having said this, neo4j doesn't really support OWL very well, though there is a semantic extension that may - you'd have to ask on their FAQ, since I haven't tried to do that.  We have users that have implemented FIBO in Allegrograph from Franz, Stardog, RDFOx, GraphDB from Ontotext, and maybe other graph databases as well, though I don't always hear about what they are doing. But you want one that supports RDF natively, even if it doesn't support all of OWL.  I think GraphDB has a free version that you should be able to load FIBO into, as one example.

I hope this is useful,

Elisa

**Comment by DennisWisnosky** [comment]
What organization are you with.

I am no longer working on FIBO.  The Enterprise Knowledge Graph Foundation
is going way beyond FIBO.

I have forwarded your email to Jacobus Geluk who will answer your
question.

Sincerely,


On Wed, May 17, 2023 at 11:03 AM rthomaskelly ***@***.***>
wrote:

> Hi I was looking into FIBO and thought it was really cool. I went ahead
> and loaded the ontology into neo4j to look and play. I wanted to make sure
> I'm doing it right? I was looking at your EquitiesExampleIndividuals file
> which has some dummy data. I tried to look at Apple only.
>
> What graph DB do you guys use? All the FIBO refs tallied to like 20,000
> nodes which was way too much for neo4j to render holistically. neo4j maybe
> doesn't render the namespaces nicely forcing ns__ onto properties? What
> else do you do with your data?
>
> Apple example data nodes as Rdfs Labels and Skos Definitions, not sure
> which is right?
>
> By Rdfs Label
> [image: GraphAppleRdfsLabel]
> <https://user-images.githubusercontent.com/24437260/239010953-c940a643-ecc8-47b4-8196-28fa9e30057b.png>
>
> By Skos Definition
> [image: GraphAppleSkosDefinition]
> <https://user-images.githubusercontent.com/24437260/239011103-f62e9ff6-c26a-48af-abf1-3d7f387fd8a8.png>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/edmcouncil/fibo/discussions/1920>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABSGKTJ7WU7TRFNYLCACE2TXGTZDHANCNFSM6AAAAAAYFJZ7KU>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 

*Dennis E. Wisnosky *
Chairman of the Board, Enterprise Knowledge Graph Foundation

Founder, Wizdom Systems, Inc.

CTO-CA (ret), DoD Business Mission Area

DWiz C 630-240-6910

**Reply by jgeluk** [reply]
I think @ElisaKendall already answered it quite well. Besides the semantic graph databases she mentioned you also have Amazon’s Neptune, Virtuoso, Oxigraph and others.
